03 Misc/06 Production Configuration/prod.webpack.config.js

```diff
@@ -1,16 +1,64 @@
+var ExtractTextPlugin = require('extract-text-webpack-plugin');
+var path = require('path');
 var webpackMerge = require('webpack-merge');
 var CompressionPlugin = require('compression-webpack-plugin');
 var commonConfig = require('./base.webpack.config.js');
 
+var basePath = __dirname;
+
 module.exports = function () {
   return webpackMerge(commonConfig, {
     devtool: 'cheap-module-source-map',
+
+    output: {
+      path: path.join(basePath, 'dist'),
+      filename: '[chunkhash].[name].js',
+    },
+
+    module: {
+      rules: [
+        {
+          test: /\.scss$/,
+          exclude: /node_modules/,
+          loader: ExtractTextPlugin.extract({
+            fallback: 'style-loader',
+            use: [
+              {
+                loader: 'css-loader',
+                options: {
+                  modules: true,
+                  localIdentName: '[name]__[local]___[hash:base64:5]',
+                  camelCase: true,
+                },
+              },
+              { loader: 'sass-loader', },
+            ],
+          }),
+        },
+        {
+          test: /\.css$/,
+          include: /node_modules/,
+          loader: ExtractTextPlugin.extract({
+            fallback: 'style-loader',
+            use: {
+              loader: 'css-loader',
+            },
+          }),
+        },
+      ]
+    },
+
     plugins: [
       new CompressionPlugin({
         asset: '[path].gz[query]',
         algorithm: 'gzip',
         minRatio: 0.8,
       }),
+      new ExtractTextPlugin({
+        filename: '[chunkhash].[name].css',
+        disable: false,
```

Comment:
Uhmmmm.

I see both ideas as good ideas.

Yours make it simpler, mine makes easier to follow.

On the one hand I like the idea of making things clear and not having to worry about "hacks" on the base config. On the other hand, yes, it is simpler.

@brauliodiez what is your vote? Nasdan idea of making it simpler? Or leaving it as is and explaining to the users this and telling them that they could use some flags to reduce code?